### PR TITLE
feat(#12): Cache changes on webserver

### PR DESCRIPTION
**Pull request by sathia27**
_Add details of your PR here.._

## Checklist
-  [x] Have you followed the [guidelines for contributing](https://github.com/javatarz/github-contribution-leaderboard/blob/master/contributing.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/javatarz/github-contribution-leaderboard/pulls) for the same feature you're adding?
- [x] Have you run tests (`pipenv run pytest`) before sending this PR?
- [x] Have you run linting (`./lint.sh`) before sending this PR?

### CODE REVIEW

**Comment by javatarz** [comment]
Hey @sathia27, thank you for your contribution! I noticed you're importing `redis` but it isn't mentioned in the `Pipfile`. The `Pipfile` is what ensures all python dependencies are installed consistently on all machines. If you run `pipenv install redis`, it'll update the Pipfile and Pipfile.lock.

This is also a good opportunity for you to send another PR to update the `contribution.md` file about this so future contributors know they should be doing this :)
I'll create an issue for this.

File: ghcl/cache.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import redis
```
Comment (javatarz) [inline ghcl/cache.py:1]: Can you run `pipenv install redis` on your machine and commit the `Pipfile` and `Pipfile.lock` as well please?

File: ghcl/cache.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    def getinstance(*args, **kwargs):
+        if class_ not in instances:
+            instances[class_] = class_(*args, **kwargs)
+        return instances[class_]
+    return getinstance
+
+
+@singleton
+class Cache:
+    CACHE_DURATION_IN_SECONDS = 60
```
Comment (javatarz) [inline ghcl/cache.py]: Can we provide this as a CLI arg through `argparse` with a default value of 60?

**Review by javatarz — CHANGES_REQUESTED** [review]

**Comment by sathia27** [comment]
@javatarz 
Thank you for your quick response

Made few other changes as you mentioned.
1. pipfile changes related to importing redis - Done
2. removed constant for cache duration. Use args instead of constant - Done
3. changes to contribution.md, regarding importing new package.
     - PR: #42

**Review by javatarz — APPROVED** [review]

**Pull request merged**